src/main.rs

```diff
@@ -42,6 +44,174 @@ impl Shell {
+                            } else if cmp == "<=" {
+                                value = left.to_num_signed() <= right.to_num_signed();
+                            } else {
+                                println!("Unknown comparison: {}", cmp);
+                            }
+                        } else {
+                            println!("No right hand side");
+                        }
+                    } else {
+                        println!("No comparison operator");
+                    }
+                } else {
+                    println!("No left hand side");
+                }
+
+                self.modes.insert(0, Mode { value: value });
+                continue;
+            }
+
+            if job.command == "else" {
+                if let Some(mode) = self.modes.get_mut(0) {
+                    mode.value = !mode.value;
+                } else {
+                    println!("Syntax error: else found with no previous if");
+                }
+                continue;
+            }
+
+            if job.command == "fi" {
+                if !self.modes.is_empty() {
+                    self.modes.remove(0);
+                } else {
+                    println!("Syntax error: fi found with no previous if");
+                }
+                continue;
+            }
+
+            let mut skipped: bool = false;
+            for mode in self.modes.iter() {
+                if !mode.value {
+                    skipped = true;
+                    break;
+                }
+            }
+            if skipped {
+                continue;
+            }
+
+            // Set variables
+            if let Some(i) = job.command.find('=') {
+                let name = job.command[0..i].trim();
+                let mut value = job.command[i + 1..job.command.len()].trim().to_string();
+
+                for i in 0..job.args.len() {
+                    if let Some(arg) = job.args.get(i) {
+                        value = value + " " + &arg;
+                    }
+                }
+
+                self.set_var(name, &value);
+                continue;
+            }
+
+            // Commands
+            let mut args = job.args.clone();
+            args.insert(0, job.command.clone());
+            if let Some(command) = commands.get(&job.command.as_str()) {
+                (*command.main)(&args, self);
+            } else {
+                self.run_external_commmand(args);
+            }
+        }
+    }
+
+    fn run_external_commmand(&mut self, args: Vec<String>) {
+        if let Some(path) = args.get(0) {
+            let mut command = process::Command::new(path);
+            for i in 1..args.len() {
+                if let Some(arg) = args.get(i) {
+                    command.arg(arg);
+                }
+            }
+            match command.spawn() {
+                Ok(mut child) => {
+                    match child.wait() {
```

Comment:
I'm not sure if using `if let` matters here, since there is also matching for `Err(e)` as well?